src/components/Header/Header.test.tsx

```diff
@@ -273,3 +291,121 @@ describe("Patron API returns wrong data", () => {
     expect(firstListElement).toHaveTextContent(/log in/i);
   });
 });
+
+describe("Patron API returns expired data, but refreshes the token successfully", () => {
+  const realGet = Cookies.get;
+
+  beforeAll(async () => {
+    Cookies.get = jest.fn().mockReturnValue(mockLoginCookie);
+
+    (global as any).fetch = jest
+      .fn()
+      // Mock the fetch API call for sitewide alerts.
+      .mockReturnValueOnce(
+        Promise.resolve({
+          status: 200,
+          json: () => Promise.resolve(refineryResponse),
+        })
+      )
+      // Mock the fetch API call for patron data (token expired).
+      .mockReturnValueOnce(
+        Promise.resolve({
+          status: 401,
+          json: () => Promise.resolve(mockExpiredResponseData),
+        })
+      )
+      // Mock the fetch API call to refresh accessToken (succeed)
+      .mockReturnValueOnce(
+        Promise.resolve({
+          status: 200,
+        })
+      )
+      // Mock the fetch API call for patron data (succeed).
+      .mockReturnValueOnce(
+        Promise.resolve({
+          status: 200,
+          json: () => Promise.resolve(mockResponseData),
+        })
+      );
+
+    await waitFor(() => render(<Header />));
+  });
+
+  afterAll(() => {
+    Cookies.get = realGet;
+    jest.clearAllMocks();
+  });
+
+  it("displays logged in view if refresh is successful", async () => {
```

Comment:
Add another test to verify all the URL endpoints that are hit.

```
expect((global as any).fetch).toHaveBeenNthCalledWith(1, alertsApiUrl);
expect((global as any).fetch).toHaveBeenNthCalledWith(2, "https://platform.nypl.org ....."); // (make the string a variable).
// etc
```

There should be a total of four calls.